src/lib.rs

```diff
@@ -529,49 +560,157 @@ pub mod pallet {
+			MembersVotes::<T>::remove_prefix((supersig_id, call_id), None);
+		}
+
+		fn unchecked_remove_supersig_from_storages(
+			supersig_id: SupersigId,
+			supersig_account: &T::AccountId,
+		) {
+			NonceCall::<T>::remove(supersig_id);
+			Members::<T>::remove_prefix(supersig_id, None);
+			TotalMembers::<T>::remove(supersig_id);
+			Calls::<T>::remove_prefix(supersig_id, None);
+			Votes::<T>::remove_prefix(supersig_id, None);
+			MembersVotes::<T>::remove_prefix((supersig_id,), None);
+
+			frame_system::Pallet::<T>::dec_consumers(supersig_account);
+		}
+
+		fn internal_add_members(
+			supersig_id: SupersigId,
+			members: BoundedVec<(T::AccountId, Role), T::MaxMembersPerTransaction>,
+		) -> Result<Vec<(T::AccountId, Role)>, Error<T>> {
+			let mut added = Vec::new();
+
+			for (member, role) in members {
+				if Self::members(supersig_id, &member) == Role::NotMember {
+					added.push((member.clone(), role.clone()));
+				}
+				Members::<T>::insert(supersig_id, member, role);
+			}
+
+			TotalMembers::<T>::try_mutate(supersig_id, |n| {
+				*n = n
+					.checked_add(added.len().try_into().map_err(|_| Error::<T>::Conversion)?)
+					.ok_or(Error::<T>::Overflow)?;
+
+				Ok(())
+			})?;
+
+			Ok(added)
+		}
+
+		#[transactional]
+		fn internal_remove_members(
+			supersig_id: SupersigId,
+			members: BoundedVec<T::AccountId, T::MaxMembersPerTransaction>,
+		) -> Result<Vec<T::AccountId>, pallet::Error<T>> {
+			let mut removed = Vec::new();
+
+			for member in members {
+				if Self::members(supersig_id, &member) != Role::NotMember {
+					Members::<T>::remove(supersig_id, member.clone());
+					removed.push(member);
+				}
+			}
+
+			TotalMembers::<T>::try_mutate(supersig_id, |n| {
+				let new_total_members =
+					n.saturating_sub(removed.len().try_into().map_err(|_| Error::<T>::Conversion)?);
+				if new_total_members < 1 {
+					return Err(Error::<T>::InvalidNumberOfMembers);
+				}
+
+				*n = new_total_members;
+
+				Ok(())
+			})?;
+
+			Ok(removed)
+		}
+
+		fn compute_deposit(data_size: usize) -> Result<BalanceOf<T>, Error<T>> {
+			let bytes_stored: u32 = data_size.try_into().map_err(|_| Error::<T>::Conversion)?;
+
+			Ok(<BalanceOf<T>>::from(bytes_stored).saturating_mul(T::DepositPerByte::get()))
+		}
+
+		fn reserve_and_record_deposit(
+			supersig_id: SupersigId,
+			supersig_account: &T::AccountId,
+			deposit: BalanceOf<T>,
+		) -> DispatchResult {
+			T::Currency::reserve(supersig_account, deposit)?;
+			TotalDeposit::<T>::try_mutate(supersig_id, |val| {
+				*val = val.checked_add(&deposit).ok_or(Error::<T>::Overflow)?;
+				Ok(())
+			})
+		}
+
+		fn unreserve_and_record_deposit(
+			supersig_id: SupersigId,
+			supersig_account: &T::AccountId,
+			amount: BalanceOf<T>,
+		) {
+			T::Currency::unreserve(supersig_account, amount);
+			TotalDeposit::<T>::mutate(supersig_id, |val| *val = val.saturating_sub(amount));
+		}
+
+		fn calc_vote_weight(supersig_id: SupersigId, who: &T::AccountId) -> Result<u32, Error<T>> {
```

Comment:
`compute_vote_weight` ? `evalutate_vote_weight` ? I think these are better names

Reply:
go for `compute_vote_weight`